python/samba/tests/smb.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+# Unix SMB/CIFS implementation. Tests for smb manipulation
+# Copyright (C) David Mulder <dmulder@suse.com> 2018
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import samba, os, random, sys
+from samba import smb
+
+PY3 = sys.version_info[0] == 3
+addom = 'addom.samba.example.com/'
+test_contents = 'abcd'*256
+utf_contents = u'Süßigkeiten Äpfel '*128
+test_literal_bytes_embed_nulls = b'\xff\xfe\x14\x61\x00\x00\x62\x63\x64'*256
+binary_contents = b'\xff\xfe'
+binary_contents = binary_contents + "Hello cruel world of python3".encode('utf8')*128
+test_dir = os.path.join(addom, 'testing_%d' % random.randint(0,0xFFFF))
+test_file = os.path.join(test_dir, 'testing').replace('/', '\\')
+
+class SMBTests(samba.tests.TestCase):
+    def setUp(self):
+        super(SMBTests, self).setUp()
+        self.server = os.environ["SERVER"]
+        creds = self.insta_creds(template=self.get_credentials())
+        self.conn = smb.SMB(self.server,
+                            "sysvol",
+                            lp=self.get_loadparm(),
+                            creds=creds)
+        self.conn.mkdir(test_dir)
+
+    def tearDown(self):
+        global test_dir
```

Comment:
I think all the `global` keyword in this file are not necessary. In Python, you only need `global` when you try to change it.